src/Microsoft.Diagnostics.Monitoring/MonitoringSourceConfiguration.cs

```diff
@@ -0,0 +1,164 @@
+                    ";ActivityId=*Activity.Id" +
+                    ";ActivitySpanId=*Activity.SpanId" +
+                    ";ActivityTraceId=*Activity.TraceId" +
+                    ";ActivityParentSpanId=*Activity.ParentSpanId" +
+                    ";ActivityIdFormat=*Activity.IdFormat" +
+                    ";ActivityId=*Activity.Id" +
+                    "\r\n" +
+                "HttpHandlerDiagnosticListener/System.Net.Http.HttpRequestOut.Stop@Activity2Stop:-" +
+                    ";ActivityDuration=*Activity.Duration.Ticks" +
+                    ";ActivityId=*Activity.Id" +
+                "\r\n" +
+
+                "\"";
+
+        public MonitoringSourceConfiguration(int metricIntervalSeconds = 60)
+        {
+            MetricIntervalSeconds = metricIntervalSeconds.ToString(CultureInfo.InvariantCulture);
+        }
+
+        protected virtual string MetricIntervalSeconds { get; }
+
+        public virtual IList<EventPipeProvider> GetProviders()
+        {
+            var providers = new List<EventPipeProvider>()
+            {
+                // Runtime Metrics
+                new EventPipeProvider(
+                    SystemRuntimeEventSourceName,
+                    EventLevel.Informational,
+                    (long)ClrTraceEventParser.Keywords.None,
+                    new Dictionary<string, string>() {
+                        { "EventCounterIntervalSec", MetricIntervalSeconds }
+                    }
+                ),
+                new EventPipeProvider(
+                    MicrosoftAspNetCoreHostingEventSourceName,
+                    EventLevel.Informational,
+                    (long)ClrTraceEventParser.Keywords.None,
+                    new Dictionary<string, string>() {
+                        { "EventCounterIntervalSec", MetricIntervalSeconds }
+                    }
+                ),
+                new EventPipeProvider(
+                    GrpcAspNetCoreServer,
+                    EventLevel.Informational,
+                    (long)ClrTraceEventParser.Keywords.None,
+                    new Dictionary<string, string>() {
+                        { "EventCounterIntervalSec", MetricIntervalSeconds }
+                    }
+                ),
+                
+                // Application Metrics
+                //new EventPipeProvider(
+                //    applicationName,
+                //    EventLevel.Informational,
+                //    (long)ClrTraceEventParser.Keywords.None,
+                //    new Dictionary<string, string>() {
+                //        { "EventCounterIntervalSec", MetricIntervalSeconds }
+                //    }
+                //),
+
+                // Logging
+                new EventPipeProvider(
+                    MicrosoftExtensionsLoggingProviderName,
+                    EventLevel.LogAlways,
+                    (long)(LoggingEventSource.Keywords.JsonMessage | LoggingEventSource.Keywords.FormattedMessage)
```

Comment:
I'd expect we wouldn't use FormattedMessage if we are also using JsonMessage, they appear duplicative?